### ISSUE TITLE
Behavior of file metadata

### ISSUE THREAD

**Issue by jdagilliland** [sbg/sevenbridges-python#1]
Even though the metadata class subclasses `dict`, the typical `dict` methods like `update` do not work with it (properly), e.g.
```
my_files = api.files.query(project=project_id, names=[fname])
my_files[0].metadata.update(new_metadata)
my_files[0].save()
print(my_files[0].metadata)
```
Does not actually save the new metadata to the platform, although the last step seems to print the updated metadata that the user wishes had gone to the platform.

My best guess is that it has to do with [the `__setitem__` method](https://github.com/sbg/sevenbridges-python/blob/3a06f295c78be8a9950d304cc9415921a4afc2e0/sevenbridges/models/compound/metadata.py#L16-L18); it seems that that method does special stuff with the metadata object (above and beyond the regular `dict` stuff), and perhaps that isn't being carried through to an `update` method call?

**Comment by SenadI** [comment]
Hi @jdagilliland 

Thanks for the issue report, nice catch. 

I guess is that trouble is that update does no use `__setitem__`, and `__setitem__` maintains set of "dirty" fields that is used to send update request to server. My first guess is that overriding `update` method and updating dirty fields would solve problem. I am working on a fix.

**Comment by jdagilliland** [comment]
Makes sense, thx!

**Comment by SenadI** [comment]
Hi @jdagilliland 

Fix for this issue is in version 0.2.0. The lib is already uploaded to pipy. If you are investigating the code please use default branch (develop) because I had to set master to be protected. Soon I will initiate a proper git glow.

Again thank you for your report.
